flow/hooks/log_operations.py

```diff
@@ -0,0 +1,146 @@
+        from flow import FlowProject
+        from flow.hooks import LogOperations
+
+
+        class Project(FlowProject):
+            pass
+
+
+        def install_operation_log_hook(operation_name, project_cls):
+            log = LogOperation(f"{operation_name}.log")
+            return lambda op: log.install_operation_hooks(op, project_cls)
+
+
+        @install_operation_log_hook("foo", Project)
+        @Project.operation(directives={
+            "submit_options": "--debug"  # Always submit operation foo with the --debug flag
+        })
+        def foo(job):
+            pass
+
+
+    The code block below provides an example of how install  :class:`~.LogOperations` to a
+    instance of :class:`~.FlowProject`
+
+    .. code-block:: python
+        from flow import FlowProject
+        from flow.hooks import LogOperations  # Import build
+
+
+        class Project(FlowProject):
+            pass
+
+
+        # Do something
+
+
+        if __name__ == "__main__":
+            project = Project()
+            project = LogOperations().install_project_hooks(project)
+            project.main()
+
+
+
+    Parameters
+    ----------
+    fn_logfile: log filename
+        The name of the log file in the job workspace. Default is "execution-record.log".
+    """
+
+    def __init__(self, fn_logfile="execution-record.log"):
+        self._fn_logfile = fn_logfile
+        # getLogger keep its own cache. This just serves to reduce the time spent setting up loggers
+        # by only doing it once.
+        self._loggers = {}
+
+    def on_start(self, operation, job):
+        """Log the start of execution of a given job(s) operation pair."""
+        self._get_logger(job).info(f"Starting execution of operation '{operation}'.")
+
+    def on_success(self, operation, job):
+        """Log the successful completion of a given job(s) operation pair."""
+        self._get_logger(job).info(
+            f"Successfully finished execution of operation '{operation}'."
+        )
+
+    def on_exception(self, operation, error, job):
+        """Log the raising of an error in the execution of a given job(s) operation pair."""
+        self._get_logger(job).info(
+            f"Execution of operation '{operation}' failed with error '{error}'."
+        )
+
+    def _get_logger(self, job):
+        if job not in self._loggers:
+            self._loggers[job] = self._setup_logger(job)
+        return self._loggers[job]
+
+    def _setup_logger(self, job):
+        logger = logging.getLogger(str(job))
+        fh = logging.FileHandler(job.fn(self._fn_logfile))
+        fh.setLevel(logging.DEBUG)
+        formatter = logging.Formatter(
+            "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+        )
+        fh.setFormatter(formatter)
+        logger.addHandler(fh)
+        return logger
+
+    def install_operation_hooks(self, op, project_cls=None):
+        """Decorate operation to install log operation to one operation in a signac-flow project.
```

Comment:
The current docstring is pretty redundant. Is this enough?
```suggestion
        """Decorate operation to log execution.
```